plugin-flex-ts-template-v2/src/feature-library/conversation-transfer/flex-hooks/components/TaskCanvasHeader.tsx

```diff
@@ -24,11 +25,13 @@ export const componentHook = function addConvTransferButtons(flex: typeof Flex)
 
   const replaceEndTaskButton = (task: ITask) => {
     if (TaskHelper.isCBMTask(task) && task.taskStatus === 'assigned') {
+      const interactionParticipants =
+        props?.store?.getState()?.[reduxNamespace]?.supervisorBargeCoach?.interactionParticipants;
       // more than two participants or are there any active invites?
       const conversationState = StateHelper.getConversationStateForTask(task);
       if (
         conversationState &&
-        (conversationState.participants.size > 2 || countOfOutstandingInvitesForConversation(conversationState))
+        (interactionParticipants > 2 || countOfOutstandingInvitesForConversation(conversationState))
```

Comment:
This would break functionality if the supervisor-barge-coach feature is removed, but not the conversation-transfer feature. This will need to be implemented a different way to not introduce a co-dependency of features.

Reply:
You could maybe approach it with the following 

1. Add a config value in conversation-transfer that checks whether supervisor barge coach is enabled or not (similar example is in custom-transfer-directory with its dependency on conversation-transfer)

2. Use that config value here to determine whether to extract the participants count from redux or use the old method

Reply:
It looks like this piece of state is only set from the conversation-transfer feature, and only read by the conversation-transfer feature, so we shouldn't need multiple implementations. This piece of state can probably just live within the conversation-transfer feature, but after I've looked into things some more, I don't think we need to keep this state at all!

I think we should remove `useParticipantCountEffect` and instead fetch the participant list using existing data in Redux, which seems to be up-to-date and includes supervisor participants. Its location in Redux is `flex.chat.conversations[conversationSid].participants`. Using this should be a lot more efficient, as the current code makes 8 or so network requests when selecting the task.